Trigger/src/EvalWeightedEventCounts_module.cc

```diff
@@ -0,0 +1,223 @@
+    kNTrackTrig = 20,
+    kNTrackTrigVar = 30,
+    kNHelixTrig = 40,
+    kNHelixTrigVar = 130,
+    kNCaloCalib = 5,
+    kNCaloCalibVar = 30,
+    kNCaloOnly = 5,
+    kNCaloOnlyVar = 30,
+    kNOcc = 100,
+    kNOccVar = 100
+  };
+
+  struct occupancyHist_ {
+    TH1F* _hOccInfo[kNOcc][kNOccVar];
+    TH2F* _h2DOccInfo[kNOcc][kNOccVar];
+
+    occupancyHist_() {
+      for (int i = 0; i < kNOcc; ++i) {
+        for (int j = 0; j < kNOccVar; ++j) {
+          _hOccInfo[i][j] = NULL;
+          _h2DOccInfo[i][j] = NULL;
+        }
+      }
+    }
+  };
+
+  explicit EvalWeightedEventCounts(const Parameters& config);
+  virtual ~EvalWeightedEventCounts() {}
+
+  virtual void beginJob();
+  virtual void endJob();
+
+  // This is called for each event.
+  virtual void analyze(const art::Event& e);
+  virtual void beginRun(const art::Run& run);
+
+  void bookHistograms();
+  void bookOccupancyInfoHist(art::ServiceHandle<art::TFileService>& Tfs, occupancyHist_& Hist);
+
+private:
+  int _diagLevel;
+  art::InputTag _evtWeightTag;
+
+  float _nProcess;
+  std::string _tagName;
+  double _nPOT;
+
+  occupancyHist_ _occupancyHist;
+
+  const mu2e::Tracker* _tracker;
+
+  // the following pointer is needed to navigate the MC truth info of the strawHits
+  const art::Event* _event;
+
+  float _minPOT, _maxPOT;
+  double _wgSum[3] = {0};
+};
+
+EvalWeightedEventCounts::EvalWeightedEventCounts(const Parameters& config) :
+    art::EDAnalyzer(config), _diagLevel(config().diagLevel()), _evtWeightTag(config().evtWeightTag()),
+    _nProcess(config().nProcess()), _tagName(config().tagName()), _minPOT(1e6), _maxPOT(4e8) {}
+
+void EvalWeightedEventCounts::bookHistograms() {
+  art::ServiceHandle<art::TFileService> tfs;
+
+  bookOccupancyInfoHist(tfs, _occupancyHist);
+}
+
+void EvalWeightedEventCounts::bookOccupancyInfoHist(art::ServiceHandle<art::TFileService>& Tfs,
+                                                    occupancyHist_& Hist) {
+
+  int index_last = 0;
+  art::TFileDirectory occInfoDir = Tfs->mkdir("occInfoGeneral");
+  Hist._hOccInfo[index_last][0] = occInfoDir.make<TH1F>(
+      Form("hInstLum_%i", index_last), "distrbution of instantaneous lum; p/#mu-bunch", 1000,
+      _minPOT, _maxPOT);
+  Hist._hOccInfo[index_last][1] = occInfoDir.make<TH1F>(
+      Form("hInstLum2B_%i", index_last),
+      "distrbution of instantaneous lum re-weighted in 2 batch-mode; p/#mu-bunch", 1000, _minPOT,
+      _maxPOT);
+}
+
+//--------------------------------------------------------------------------------//
+void EvalWeightedEventCounts::beginJob() { bookHistograms(); }
+
+//--------------------------------------------------------------------------------//
+void EvalWeightedEventCounts::endJob() {
+
+  //    evalTriggerRate();
+  printf("[%s::enJob] totWg    = %10.3e\n", _tagName.c_str(), _wgSum[0]);
+  printf("[%s::enJob] totEvtB1 = %10.3f\n", _tagName.c_str(), _wgSum[1]);
+  printf("[%s::enJob] totEvtB2 = %10.3f\n", _tagName.c_str(), _wgSum[2]);
```

Comment:
Why not print the name of the input tag?  That reduces the number of things to maintain - imagine that someone changes just the input tag but not the tag; if you look at the log file there is no way to know which is which without consulting how the file was made.

Reply:
Discussed with Giani and decided that it's possibly better to let people supply the tags and be consistent in the fcl, as this can also allow people to include multiple trigger lines together, e.g. a tag called "LowP" (includes both low momentum Cpr and Tpr).

Reply:
OK - I can see that as a reasonable future development.